spec/rubocop/cop/rails/unused_ignored_columns_spec.rb

```diff
@@ -111,13 +111,41 @@ class User < ApplicationRecord
         expect_offense(<<~RUBY)
           class User < ApplicationRecord
             self.ignored_columns += ['real_name']
-                                     ^^^^^^^^^^^ Remove `real_name` from `ignored_columns` because the column does not exist.
+                                     ^^^^^^^^^^^ Remove `real_name` from `ignored_columns` because the column does not exist in schema `db/schema.rb`.
           end
         RUBY
       end
     end
   end
 
+  context 'when setting SchemaPath' do
+    include_context 'with SchemaLoader'
+
+    let(:cop_config) do
+      {
+        'SchemaPath' => 'db/other_schema.rb'
+      }
+    end
+    let(:schema) { <<~RUBY }
+      ActiveRecord::Schema.define(version: 2020_02_02_075409) do
+        create_table "users", force: :cascade do |t|
+          t.string "account", null: false
+        end
+      end
+    RUBY
+
+    it 'registers an offense to the nonexistent column' do
+      allow(RuboCop::Rails::SchemaLoader).to receive(:load).and_call_original
+      expect_offense(<<~RUBY)
+        class User < ApplicationRecord
+          self.ignored_columns += ['real_name']
+                                   ^^^^^^^^^^^ Remove `real_name` from `ignored_columns` because the column does not exist in schema `db/other_schema.rb`.
+        end
+      RUBY
+      expect(RuboCop::Rails::SchemaLoader).to have_received(:load).with(anything, 'db/other_schema.rb').at_least(:once)
+    end
+  end
```

Comment:
My assumption is that if the `SchemaPath` is set, the underlying `SchemaLoader` code will load the file at that path. This is why I did not add more examples in this RSpec context, but if that's a bad assumption, I can certainly attempt to add more here.